test/test_backend/routes/test_graph.py

```diff
@@ -0,0 +1,390 @@
+single 30000,antibiotics 1,909552172,180
+single small,single-small,1000002363,30
+single small,single-small-second-time,1000002363,30
+"""
+TEST_CASES_HOLD = """
+testType,testName,codeset_ids,timeoutSeconds
+"""
+TEST_CASES = TEST_CASES_FAST
+EXTRA_CONCEPT_IDS = []
+HIDE_VOCABS = ['RxNorm Extension']
+
+
+class TestGraph(unittest.TestCase):
+    """Tests for graph.py"""
+    test_cases: List[Dict] = []
+
+    @staticmethod
+    def _get_test_cases() -> List[Dict]:
+        """Set up"""
+        csv_file = StringIO(TEST_CASES_FAST.strip())
+        cases: List[Dict] = list(csv.DictReader(csv_file, delimiter=',', quotechar='"'))
+        for case in cases:
+            for k, v in case.items():
+                if k in ['codeset_ids', 'hide_vocabs']:
+                    case[k] = v.split(',') if v else []
+        return cases
+
+    @staticmethod
+    def _save_output(
+        test_name: str, sg: DiGraph, nodes_in_graph: Set[int], preferred_concept_ids: Set[int],
+        orphans_not_in_graph: Set[int], hidden: Dict[str, Set[int]], minify_threshold=1000, base_case=False
+    ):
+        """Save outputs
+        :param minify_threshold: If the number of records in the JSON is greater than this, will save minified, else
+        will use indent of 4.
+        :param base_case: If True, this will be considered the new base case for testing against in the future, and
+        outputs will be saved not to a hash subfolder, but to STATIC_DIR_get_connected_subgraph. For reference, the hash
+         will be saved in _base_case_hash.txt."""
+        # Find commit hash & determine how to use it
+        hash_proc = subprocess.run(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE, text=True, check=True)
+        git_hash = hash_proc.stdout.strip()[0:7]
+        subdir = STATIC_DIR_get_connected_subgraph
+        subdir = subdir / f'{git_hash}' if not base_case else subdir
+        subdir.mkdir(parents=True, exist_ok=True)
+        path = subdir / f'{test_name}.json'
+        if base_case:
+            with open(subdir / '_base_case_hash.txt', 'w') as f:
+                f.write(git_hash)
+        # Format & save output
+        normalized_hidden = {k: sorted(list(v)) for k, v in hidden.items()}
+        results = {
+            'sg.edges': sorted(list(sg.edges)),
+            'nodes_in_graph': sorted(list(nodes_in_graph)),
+            'preferred_concept_ids': sorted(list(preferred_concept_ids)),
+            'orphans_not_in_graph': sorted(list(orphans_not_in_graph)),
+            'hidden': normalized_hidden,
+        }
+        n_records = sum([len(v) for k, v in results.items()])
+        indent = None if n_records > minify_threshold else 4
+        with open(path, 'w') as f:
+            f.write(json.dumps(results, indent=indent))
+
+    # todo: this is not currently used. It needs modification after changes to get_connected_subgraph()
+    async def _create_inputs__get_connected_subgraph__needs_repair(self):
```

Comment:
## Drop `async`
siggie async def don't run in normal suite, else if run manually they just pass without doing anything
Apparently with `async`, these don't run in the normal test suite when called.

I think if you still need to run these `async`, we should split most of the test into a separate function. Hopefully that way (i) you can call it when you need, as async, and (ii) it will run via the test suite.

Reply:
obsolete. i have since fixed async to work with tests